docs/tasks/access-kubernetes-api/extend-api-custom-resource-definitions.md

```diff
@@ -0,0 +1,158 @@
+a new API object type. The new API object type will be given an API endpoint
+URL and support CRUD operations, and watch API. You can then create custom
+objects using this API endpoint. You can think of `CustomResourceDefinitions`
+as being much like the schema for a database table. Once you have created the
+table, you can then start storing rows in the table. Once created,
+`CustomResourceDefinitions` can act as the data model behind custom controllers
+or automation programs.
+
+A `CustomResourceDefinition` creates the REST API for a `CustomResource` of your chosen name.
+
+## Creating a CustomResourceDefinition
+
+When you create a new `CustomResourceDefinition`, the Kubernetes API Server
+reacts by creating a new RESTful resource path (namespaced or cluster-scoped)
+depending on your request.  As with existing built-in objects, deleting a
+namespace deletes all custom objects in that namespace.
+`CustomResourceDefinitions` themselves are non-namespaced and are available to all namespaces.
+
+For example, if you save the following `CustomResourceDefinition` to `resourcedefinition.yaml`:
+
+```yaml
+apiVersion: apiextensions.k8s.io/v1beta1
+kind: CustomResourceDefinition
+metadata:
+  # name must be in the form: plural.group
+  name: crontabs.stable.example.com
+spec:
+  # group name to use for REST API: /apis/<group>/<version>
+  group: stable.example.com
+  # version name to use for REST API: /apis/<group>/<version>
+  version: v1
+  # either Namespaced or Cluster
+  scope: Namespaced
+  names:
+    # plural name to be used in the URL: /apis/<group>/<version>/<plural>
+    plural: crontabs
+    # singular name to be used as alias on the CLI and for display
+    singular: crontab
+    # kind is normally the CamelCased singular type. Your resource manifests use this
+    kind: CronTab
+    # shortNames allow shorter string to match your resource on the CLI
+    shortNames:
+    - ct
+```
+
+And create it:
+
+```shell
+$ kubectl create -f resourcedefinition.yaml
+customresourcedefinitions "crontabs.stable.example.com" created
+```
+
+Then a new RESTful API endpoint is created at:
+
+`/apis/stable.example.com/v1/namespaces/<namespace>/crontabs/...`
+
+This endpoint URL can then be used to create and manage custom objects.
+The `kind` of these objects will be `CronTab` from the spec of the
+`CustomResourceDefinition` object we created above.
+
+
+## Creating Custom Objects
+
+After the `CustomResourceDefinition` object has been created you can create
+custom objects. Custom objects can contain custom fields. These fields can
+contain arbitrary JSON. 
+In the following example, a `cronSpec` and `image` custom fields are set to the
+custom object of kind `CronTab`.  The kind `CronTab` comes from the spec of the
+`CustomResourceDefinition` object we created above.
+
+If you save the following YAML to `my-crontab.yaml`:
+
+```yaml
+apiVersion: "stable.example.com/v1"
+kind: CronTab
+metadata:
+  name: my-new-cron-object
+cronSpec: "* * * * /5"
```

Comment:
Not for every resource a spec makes sense. Compare `Endpoints`. I don't see that we need this, even for a later status subresource.